src/System/Terminal/Widgets/Select.hs

```diff
@@ -34,24 +34,44 @@ instance Widget Select where
                     , " "
                     , t
                     ]
-         in Text.unlines $ s.prompt : (mkOption <$> s.options)
+            prompt = s.prompt <> ":"
+         in Text.unlines $ prompt : (mkOption <$> s.options)
     handleEvent (KeyEvent (ArrowKey Upwards) []) = moveUp
     handleEvent (KeyEvent (ArrowKey Downwards) []) = moveDown
-    handleEvent (KeyEvent SpaceKey []) = filtered (not . (.multiselect)) %~ unsetAll >>> flipCurrent
+    handleEvent (KeyEvent SpaceKey []) = \s@(Select{multiselect}) ->
+        if not multiselect
+            then flipCurrent $ unsetAll s
+            else s
```

Comment:
Let's use guards here.

Reply:
due to the lambda, it would only be possible to achieve guard-like syntax with -XMultiwayIf, but i dont image thats what you want. The lambda is required so long as the equations bind a single argument

Reply:
So let’s not bind a single argument. Without lens, the lambda is strictly worse than taking all the params IMO.